### PR TITLE
Add new Universe and CoordTransformer

### PR DESCRIPTION
**Pull request by Capital-Asterisk**
The old universe and coordinate space code was incredibly over-complicated: a prematurely generalized mess of templates and type erasure. 

Most of the new universe structures are not usable yet, and aren't final.

Prioritizing the difficult parts first, I added the CoordTransformer and a unit test. Aside from the all algebra (and the diff turning out quite horribly), it should be fairly straightforward.

## New Universe differences

The main thing I'm changing is that I'm deciding to handle Satellite IDs as local to their coordinate-space, instead of being global.

In the old universe, a Satellite taking off from a planet surface and into deep space would have kept the same ID throughout all the coordinate spaces it may transfer between. In this new system, the Satellite's data will be moved between each coordinate space it transfers to, given a new ID each time.

This results in improved isolation between coordinate spaces, such as allowing Satellites to be created without bothering the rest of the universe (this was a problem before).

### CODE REVIEW

File: src/osp/universe/coordinates.h [hunk 1]
```diff
@@ -24,316 +24,188 @@
...
-        // Make sure all CComps are valid
-        if ( ! (m_components.at(ccomp_id<CCOMP_T>).has_value() && ...))
-        {
-            return std::nullopt;
-        }
-
-        return TupleCComp_t<CCOMP_T ...>(
-                    Corrade::Containers::arrayCast<typename CCOMP_T::datatype_t>(
-                                        m_components.at(ccomp_id<CCOMP_T>).value()) ...);
+        return m_n == 0 && m_c.isZero();
```
Comment (jonesmz) [inline src/osp/universe/coordinates.h:95]: parans around parts of the bool logic

File: src/osp/universe/coordinates.h [hunk 2]
```diff
@@ -24,316 +24,188 @@
...
+        CoSpaceTransform const& parent, CoSpaceTransform const& child) noexcept
 {
-    int16_t exp = parentPrec - childPrec;
-    // individual vector components is workaround for non-constexpr operator-()
-    return CoordspaceTransform{exp, CoordspaceTransform::Vec_t( -childPos.x(), -childPos.y(), -childPos.z()), exp};
+    int const precisionDiff = child.m_precision - parent.m_precision;
+    return {
+        .m_rotate   = child.m_rotation,
+        .m_c        = child.m_position,
+        .m_n        = -precisionDiff,
```
Comment (jonesmz) [inline src/osp/universe/coordinates.h:206]: minor, use std::negate() over simply "-". While they do the same thing, std::negate is much harder to miss when casually reading.

File: test/universe/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+##
+PROJECT(test_universe CXX)
+ADD_TEST_DIRECTORY(${PROJECT_NAME})
+
+TARGET_LINK_LIBRARIES(test_universe longeron EnTT::EnTT Corrade::Main Magnum::Magnum)
```
Comment (jonesmz) [inline test/universe/CMakeLists.txt:28]: doesn't Magnum::Magnum pull in Corrade::Main anyway?

**Review by jonesmz — APPROVED** [review]

**Comment by Capital-Asterisk** [comment]
Will fix some of these in an upcoming pull request

**Pull request merged**